GrandCentralBoard/Widgets/Harvest/BillableDates.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  BillableDates.swift
+//  GrandCentralBoard
+//
+//  Created by Maciek Grzybowski on 03.06.2016.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import Foundation
+
+
+final class BillableDates {
+
+    let dates: [NSDate]
+
+    init(referenceDate: NSDate, numberOfPreviousDays: Int, includeWeekends: Bool?) {
+        let calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierGregorian)!
+        let sunday = 1
+        let saturday = 7
+
+        var previousDays: [NSDate] = []
+        var date = referenceDate
+
+        while previousDays.count < numberOfPreviousDays {
```

Comment:
The problem with using  sequence is that getting the number of elements explicitly is hard. If `includeWeekends` is set to `false` and I use gregorian calendar, the sequence from `1` to `numberOfPreviousDays` won't be sufficient, because some dates will be filtered out as being a weekend. Thus, I'd need to compensate for them, by expanding the sequence before filtering... and so on... And how to deal with other calendars? 😰 

Hence, IMO doing implicit `while` loop is much simpler. Or do you have any good idea for handling `includeWeekends = false` in a sequence style iteration?